### PR TITLE
ECO-522 auto restart server

### PR DESCRIPTION
**Pull request by liorama**

### CODE REVIEW

File: scripts/src/metrics.ts [hunk 1]
```diff
@@ -45,6 +45,10 @@ export function reportServerError(method: string, path: string) {
 	statsd.increment("server_error", 1, undefined, { method, path });
 }
 
+export function reportProcessAbort() {
```
Comment (doodyparizada) [inline scripts/src/metrics.ts]: maybe attach a reason instead of system: exit? i.e. reason: "multiple 5xx errors in short period"

Comment (doodyparizada) [reply scripts/src/metrics.ts]: the reason can be a parameter

File: scripts/src/middleware.ts [hunk 2]
```diff
@@ -8,8 +8,14 @@ import { getConfig } from "./config";
 import { generateId, pick } from "./utils";
 import { MarketplaceError } from "./errors";
 import { getDefaultLogger } from "./logging";
+import { abort as restartServer } from "./server";
 
 const START_TIME = (new Date()).toISOString();
+
+const RESTART_ERROR_COUNT = 5;  // Amount io errors to occur in time frame to trigger restart
```
Comment (doodyparizada) [inline scripts/src/middleware.ts]: why io errors? can't it be any unknown error that causes 5xx?

File: scripts/src/middleware.ts [hunk 3]
```diff
@@ -113,9 +119,12 @@ function clientErrorHandler(error: MarketplaceError, req: express.Request, res:
 
 function serverErrorHandler(error: any, req: express.Request, res: express.Response) {
 	const log = req.logger || logger;
-
 	metrics.reportServerError(req.method, req.url);
 
+	const timestamp = Math.floor(Date.now() / 1000);
```
Comment (doodyparizada) [inline scripts/src/middleware.ts]: `moment.utc().seconds()` or  `new Date().getUTCSeconds();`

File: scripts/src/middleware.ts [hunk 4]
```diff
@@ -129,6 +138,12 @@ function serverErrorHandler(error: any, req: express.Request, res: express.Respo
 		message += `message: ${ error.toString() }`;
 	}
 
+	if (serverErrorTimeStamps.length === RESTART_ERROR_COUNT) {
```
Comment (doodyparizada) [inline scripts/src/middleware.ts]: if you initialize `serverErrorTimeStamps` to 20 zeros, you can remove this `if`

File: docker-compose.dev.yaml [hunk 5]
```diff
@@ -5,6 +5,7 @@ services:
   marketplace-server:
     image: kinecosystem/marketplace-server:latest
     command: ./tests/wait-for marketplace-internal:3000 -- npm run start
+    restart: no
```
Comment (doodyparizada) [inline docker-compose.dev.yaml]: `no` translate to `false` in yaml, so you need parenthesis `"no"`. That's why travis is failing

File: scripts/src/middleware.ts (hunk 2 above)
Comment (liorama) [reply scripts/src/middleware.ts]: lol, spelling mistake. suppose to be "of"

File: scripts/src/middleware.ts (hunk 3 above)
Comment (liorama) [reply scripts/src/middleware.ts]: Both are not good, they give the seconds value of the current time. So you are limited to a timeframe lower than a minute and if more than a minute passes you got a bug on your hands.
If anything `moment().unix()` fits better.

File: scripts/src/middleware.ts (hunk 4 above)
Comment (liorama) [reply scripts/src/middleware.ts]: We can minify the code if we just want to make it less readable.

**Comment by doodyparizada** [comment]
looks good, but travis failed. maybe @nitzantomer can take a look?

**Pull request merged**